src/PublicApiAnalyzers/UnitTests/AnnotatePublicApiAnalyzerTests.cs

```diff
@@ -60,6 +61,134 @@ private async Task VerifyAdditionalFileFixAsync(string source, string oldShipped
 
         #region Fix tests
 
+        [Fact, WorkItem(4040, "https://github.com/dotnet/roslyn-analyzers/issues/4040")]
+        public async Task NoObliviousWhenUnannotatedClassConstraint()
+        {
+            var source = @"
+#nullable enable
+public class C<T> where T : class
+{
+}
+";
+
+            var shippedText = @"";
+            var unshippedText = @"#nullable enable
+C<T>.C() -> void
+C<T>
+";
+
+            await VerifyCSharpAsync(source, shippedText, unshippedText, System.Array.Empty<DiagnosticResult>());
```

Comment:
I am curious on the why you are passing the empty array. Is it for clarity? Allocation? Something else?

Reply:
Will remove. Thanks